example/trace/otlpgrpc/client/client.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 OpenTelemetry Authors
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/baggage"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+	"go.opentelemetry.io/otel/trace"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/oauth"
+
+	gcppropagator "github.com/GoogleCloudPlatform/opentelemetry-operations-go/propagator"
+)
+
+func initTracer() (func(), error) {
+	ctx := context.Background()
+
+	creds, err := oauth.NewApplicationDefault(ctx)
+	if err != nil {
+		panic(err)
+	}
+	
+	exporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithDialOption(grpc.WithPerRPCCredentials(creds)))
+	if err != nil {
+		panic(err)
+	}
+
+	tp := sdktrace.NewTracerProvider(
+		// For this example code we use sdktrace.AlwaysSample sampler to sample all traces.
+		// In a production application, use sdktrace.ProbabilitySampler with a desired probability.
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
+		sdktrace.WithBatcher(exporter))
+
+	otel.SetTracerProvider(tp)
+	return func() {
+		err := tp.Shutdown(context.Background())
+		if err != nil {
+			fmt.Printf("error shutting down trace provider: %+v", err)
+		}
+	}, nil
+}
+
+func installPropagators() {
+	otel.SetTextMapPropagator(
+		propagation.NewCompositeTextMapPropagator(
+			// Putting the CloudTraceOneWayPropagator first means the TraceContext propagator
+			// takes precedence if both the traceparent and the XCTC headers exist.
+			gcppropagator.CloudTraceOneWayPropagator{},
+			propagation.TraceContext{},
+			propagation.Baggage{},
+		))
+}
+
+func main() {
+	installPropagators()
+	shutdown, err := initTracer()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer shutdown()
+	tr := otel.Tracer("cloudtrace/example/client")
+
+	client := http.Client{Transport: otelhttp.NewTransport(http.DefaultTransport)}
```

Comment:
Any reason why you are using otelhttp here, rather than just sending a simple span?

Reply:
was just copying from the existing example. I simplified it down now